Pirate Campaign - Smuggling Intro

Shroomicide opened the pull request:
This PR adds the intro arc for a planned Pirate / Smuggling campaign. The player has a small chance to be approached by the representative of a small courier firm within the year 3013 (approx. 2 months). As the player progresses, they discover that the cargo in question is not strictly legal...

In terms of content, this adds 1 introductory conversation, 5 missions, and a contingency if the player aborts, fails, or rejects too many missions.

Any and all constructive critique is welcomed! The dialogue has gone through a couple revisions, and I've tested each mission a few times to get the flow right, but I'd like to make sure it is as polished as possible.

It is also important to mention that, for the sake of player experience, the missions are not labelled as "illegal." This has been brushed aside via ambiguity, but the primary reason is to avoid putting new players in an uncomfortable situation.

Edit: Also, this is my first PR, so if I set something up wrong, I'm all ears. My pride as a programmer may be wounded, but at least my technical skills will be saved.

calniotrey commented on pirate missions smuggling intro.txt:
```diff
@@ -0,0 +1,1216 @@
+	invisible
+	landing
+	
+	to offer
```
Maybe add a "unpaid mortgages" > 0 to check if the player has loans (thanks Fzzr and Janaszar from Discord for that condition)

Shroomicide replied:
@calniotrey I was looking for something like that on the Wiki! It's good to know it exists. I hadn't seen any documentation on it. It's been added now, thanks!

calniotrey commented on pirate missions smuggling intro.txt:
```diff
@@ -0,0 +1,1216 @@
+	
+	to offer
+		#random < 5
+		"year" == 3013
```
Oh nevermind, I think you just put these condition because you commented the random. In the "final" form of the PR (ie just before merging), only the random part will remain right ?

jafdy replied:
You should put it in the final form, the people likely to be reviewing this are capable of making the required changes if they want to force a trigger.

Shroomicide replied:
@jafdy Noted; "random" has been uncommented.

Shroomicide commented on pirate missions smuggling intro.txt:
```diff
@@ -0,0 +1,1216 @@
+			#	Else, if the player started with an interceptor...
+			branch branch2
+				"ships: Interceptor" == 1
+			#	Otherwise, the player started with a shuttle.
```
No, you're not missing anything. I didn't add it at the time, because I thought it would be very unlikely for the player to acquire a second ship within two months, but I may just have to simplify that line. The branching for testing multi-ownership would be painful.

I wish I could just check if the player was in a starting ship, but I don't think it's possible. I don't know; how likely is it that the player will ditch one of the starting ships entirely? In your opinion, is it worth cutting out and just replacing the dependent dialogue with "Brand new starship?"

Moomonster commented:
As a lady, I find this game in dire need of some casual romance. Every other encounter is so "Captain Kirk". We need more Riker moments like this! 

Seriously though, I think this is quite well-written (and I really do appreciate the flirtatious options - my alter ego is a huge flirt, it's just good fun!). Seems like an easy-to-complete yet humorous intro mission. Not groan worthy like a lot of beginner missions. I would have fun doing this one, so.... props.

Pointedstick commented on data/pirate missions smuggling intro.txt:
I feel like this could use a second dialogue option for your captain.

Pointedstick commented on data/pirate missions smuggling intro.txt:
Ditto

Pointedstick commented on data/pirate missions smuggling intro.txt:
I'm not sure why your captain would ask this. Given that weapons are legal to purchase and own in the Endless Sky world, they obviously need to be transported to the outfitter by *someone.* Also "dueling sabers" are already listed among the commodities lists so clearly some weapons are legal to transport.

Maybe change the phrasing to something like "Isn't it illegal for private couriers to transport weapons?" or "It's illegal to transport anything more dangerous than class III weapons; what class do these fall under?"

Pointedstick commented on data/pirate missions smuggling intro.txt:
"larger, balding" -> "larger balding"

Pointedstick commented on data/pirate missions smuggling intro.txt:
I don't like how he immediately comments on your tone. Feels weird and forced.

Pointedstick commented on data/pirate missions smuggling intro.txt:
let -> lets

Pointedstick commented on data/pirate missions smuggling intro.txt:
> but charming!

Isn't she pretty sick of your groanworthy pick-up lines at this point?

Shroomicide replied:
Oh, certainly. She was probably sick of them to begin with. But it's also her job to respond somewhat positively to even the worst of pick-up lines; she is attempting to bamboozle the player into smuggling, after all. The best way to keep a flirtatious captain on the hook is to make them feel like they're making progress!

Pointedstick commented on data/pirate missions smuggling intro.txt:
I feel like another response that leads to the same place might fit here.

Shroomicide replied:
Agreed! I realized as I was revising it the first time that I had a few "lone" `choice` labels. This is a great place for a second one.

Pointedstick commented on data/pirate missions smuggling intro.txt:
>  To tell you the truth, I don't even smoke cigars.

> the end of his cigarette

:musical_note:  One of these things is not like the other :musical_note:

Pointedstick commented on data/pirate missions smuggling intro.txt:
> "...I think you're smarter than that."

All the more reason why "Etarip Enterprises" doesn't fit. :)

Shroomicide replied:
Haha; I absolutely agree. It started out as a silly idea, and is probably best suited for a much younger age bracket.

Pointedstick commented on data/pirate missions smuggling intro.txt:
Why not an actual Bounty Hunter fleet?

Shroomicide replied:
I honestly didn't know there was a defined bounty hunter fleet. But you're right - a bounty hunter fleet does seem slightly more appropriate for bounty hunting...

Amazinite commented:
Just a heads up, this seems like the same situation as #3974, ref: https://github.com/endless-sky/endless-sky/pull/3974#issuecomment-457893972.

>The FW intro has 30-something missions, and right now you have [five]. :)
>If your intent is to create an entire campaign, then a PR isn't the place to be doing development. Development should happen on a fork or separate repository say with help from people on Discord. A PR is for when you have a finished product that you're getting final feedback on.

Referencing #4089, even if we were to accept intros of campaigns before the rest of the campaign is finished, a PR should consist of a finished intro, or at least say a >90% finished intro.

Shroomicide commented:
@Amazinite This is actually intended to be a finished intro. The length and gameplay is substantially longer than the referenced PR (~10x the size, and roughly 60% the length of the FW intro). At a relaxed pace I'd wager it shakes out to 45 minutes. But I didn't see a good reason to make this section longer. It feels about right to me.

I'm not sure what the context of Zitchas' PR'd Remnant missions is, but it's a similar amount of content. That was my initial barometer for this submission, though I realize it's not a full campaign. I did make sure to shop it around on the Discord a few times as part of my own in-dev repository (https://github.com/Shroomicide/ES-Pirate-Campaigns), but the response was not nearly as timely or comprehensive as I've received here - which is actually why I submitted in the first place.

But yeah, I'm more than happy to hold off until it's complete. My main concern is getting quality feedback and making sure the individual pieces are of good quality and conform to the standard without needing to scrap whole subsequent sections for one reason or another.

Pointedstick commented:
I agree that this is a suitable intro all on its own. It definitely reads and plays that way.

Shroomicide commented on data/pirate missions smuggling intro.txt:
```diff
@@ -0,0 +1,1217 @@
+			`	Her smile hangs for a moment, then turns to a sigh. "It's a real shame though, making new captains into paupers, just so the banks can make themselves rich off other people's hard work."`
+			choice
+				`	"I'm not sure about that. Banks are often made out to be the enemy, but historically speaking, banks have allowed private business to flourish by providing security, allowing for collective funding, and giving the average person a safety net without relying on the government or communal goodwill. Even if they do take certain excesses and breed financial corruption when not kept in check, th ...
+				`	"BURN THE BANKS!"`
```
Oh, absolutely! I just liked the juxtaposition there - no political leanings intended. But, to balance it out, I've added a respectable criticism of finance capitalism!

Shroomicide commented on data/pirate missions smuggling intro.txt:
I was wondering where the term came from. I searched through a bunch of the mission files for reference to a portable data device to see if it had been defined, but I couldn't find any. "Padd" must have just leaped in there. Anyway, great catch - I've adjusted it accordingly!

Pointedstick commented:
I see a bunch of conversations marked as resolved by no new commits. Are you still working on it?

Shroomicide commented:
Yeah, sorry about that - I was checking them off as I finished them, and ended up running out of time. I'm throwing the new commit out there, but there are still a few edits to be made.

Shroomicide commented:
Also, I realized I commented on a bunch of change requests, but never directly responded to your first comment. Thank you so much for checking it out! I really appreciate the feedback.

Pointedstick commented:
Thanks. The worry I have with the `question` / `question2` split is that it makes editing and correcting much more error-prone in the future. I feel there must be a way to tweak the structure a bit so that the original version suffices for both. Also, you can avoid repeating yourself so much by taking advantage of the nifty new system added here: https://github.com/endless-sky/endless-sky/commit/1e2e6bd91c545e7e37d6b6ffd9dd41ebe0d75795

Also, can you clarify why there's `./data/pirate missions smuggling intro.txt` and also `./pirate missions smuggling intro.txt`?

Pointedstick commented on data/pirate missions smuggling intro.txt:
```diff
@@ -0,0 +1,1232 @@
+			choice
+				`	"I'm not sure about that. Banks are often made out to be the enemy, but historically speaking, banks have allowed private business to flourish by providing security, allowing for collective funding, and giving the average person a safety net without relying on the government or communal goodwill. Even if they do take certain excesses and breed financial corruption when not kept in check, th ...
+				`	"You know, I think you're right. No matter how I look at it, the rise of centralized finance has led to an irrevocable disparity in individual wealth. Look at all the great financial disasters of the past - before 'credit' existed, wealth could only be measured locally. Loss of wealth in one region was balanced by gain in another. Now, how much is a building really 'worth'? How about a star ...
+					goto disparity
```
It doesn't look like there's a `disparity` label.

Shroomicide commented:
Yes! That's a fantastic solution; I'm implementing it now. I was wondering if tehhowch had finished that. It's a brilliant architecture for mission writing - it makes things so much more organic, especially for reloads / multiple playthroughs.

As for the file, I have no clue how that got there. I don't recall adding it, and it isn't set up that way on my desktop GitHub workspace. I'll clean it out as soon as I can.

Shroomicide commented:
For now, I'm just going to cut the "questions2" and have it direct back for "questions" as suggested. I've also solidified the next arcs, so I'm adding a "completion" for the intro's finale. I'm uploading the current file, but I'll have a new version tomorrow.

Shroomicide commented on data/pirate missions smuggling intro.txt:
I absolutely agree! I'm at a loss for how it ended up like that..

Shroomicide commented:
That's bizarre. The only code I've added is `"unpaid mortgages" > 100000`, which has supposedly been a valid "pilot file" condition for months. Maybe it doesn't work.

Agreed. I didn't want to make it too common, but perhaps 10% would be more reasonable. Assuming an average of 10 landings within 45 days, it should offer ~65% of playthroughs.

Pointedstick commented:
I'm still not seeing the mission appear as an available option in my save files. :(

Shroomicide commented:
That's frustrating. I started through the missions with this exact fileset. Arachi was actually having the same problem a few weeks ago, but another guy got it started with the same file. To make matters worse, `random` doesn't need to be encapsulated (per existing missions), so that isn't the issue either.

This is a needless suggestion, but I'm guessing you also have the latest build?

Pointedstick commented:
Yes, I am checking out your branch and compiling everything from source.

Shroomicide commented:
That really stinks. I yanked the whole fileset (pre-merge), added the libraries and the .exe, and got the mission right away. I have no idea what could be going wrong.

Pointedstick commented:
Does it work for anyone else? Am I just smoking crack here?

comnom commented:
>Does it work for anyone else? Am I just smoking crack here?

It offers for me as is. It took a new pilot until Dec 7 to get it however, so I'd recommend just removing the random condition to test it.

@Shroomicide If this is meant to be the start of a full fledged campaign, you may consider a condition set that will always offer the start of the chain at some point. If I were doing a pirate play through, I wouldn't want to start a new pilot to only have a chance at playing the campaign.

Pointedstick commented on data/pirate missions smuggling intro.txt:
This makes the mission impossible to complete, because the destination planet is always between 6 and 10 jumps away, but you're only given two days to get it there.

jafdy replied:
Unless you get really lucky with your jumpdrive routes.

Shroomicide replied:
So, that's my bad - I had an old copy of the CreatingMissions wiki page (I do a lot of writing offline), which stated that `"a 'deadline 2' statement means offer the default deadline, plus two days"`. The current wiki clarifies this, but I never bothered to check it. That's why every mission has this bug; Alexandra wasn't lying about cutting costs by having the fastest couriers in the 'verse.

Pointedstick commented on data/pirate missions smuggling intro.txt:
I would rewrite this sentence as follows:

`"The difference is, we have reliable couriers, so we can cut expenses. We pay the difference to couriers like you, so you can get out of debt immediately - no loan interest spoiling every delivery you make`

Pointedstick commented on data/pirate missions smuggling intro.txt:
I would rewrite this a bit as follows:

`As you can see, by spreading out our network of couriers, the average time from request to delivery is cut by over 40%. Combined with an almost zero-tolerance policy for abandoned cargo, high-end clients are willing to pay hefty sums for the faster and more reliable service we provide.`


and then
`which allows couriers` -> `which nets our couriers` (or something like that)

Pointedstick commented on data/pirate missions smuggling intro.txt:
Ditto.

Pointedstick commented on data/pirate missions smuggling intro.txt:
Ditto

Pointedstick commented on data/pirate missions smuggling intro.txt:
Ditto

Shroomicide commented:
@comnom I agree! I didn't want it to be too common to start out with, as it has the potential to tie up new players, but I think I can create a "bridge" mission for players who have gotten further. I'm not sure if it would skip the intro entirely or just the first conversation regarding debt. If the latter, they could then do the 5 runs for pay. Perhaps it would have a small chance of offering if the player had picked up any of the random "illegal" spaceport jobs.

@Pointedstick And to me, apparently! I can't believe I left those bad `deadlines` in.

I fixed up the file and added a "complete" conversation to the finale, which will branch into three separate (do-in-any-order) smuggling arcs for the FW Civil War, the Republic (regional smuggling, including the Syndicate), and organized Pirate groups. The conversation is a little lengthy, but most of it can be skipped by players who do not want to read the flavor text. It should fire on an existing save as long as the final mission hasn't been started.

jafdy commented:
None of the missions can offer without the intro being offered first.

Shroomicide commented:
Right; but for testers, the final conversation should work as long as the mission hasn't been offered already. In response to comnom's point, since the entire intro is predicated on the player being in debt, I would have to either make copies of these missions with a different structure (for pay) or have more experienced pilots have an abbreviated, alternate intro. I'm certainly open to suggestions on that front!

Amazinite commented:
@Shroomicide Have you been making any progress on the rest of a pirate campaign? I'm not really keen on the idea of merging an intro for a pirate campaign if the rest of it isn't in sight. (That's one of the big arguments against merging intros before the rest of the campaign; #4089.) I think that campaigns would best be merged in full so that the same vision molds the campaign throughout, instead of having the potential situation of someone creating and getting an intro merged, then leaving that as something that the next person needs to work off of when they might not have the same vision and direction as what the intro would suggest.

TheMarksman-ES commented on data/pirate missions smuggling intro.txt:
`log "People" "Alexandra Merrin" `Alexandra Merrin is a representative of a courier firm named 'Liekedeeler Enterprises,' one that helps new pilots pay off their loans in exchange for a rather lucrative contract agreement.`
`

TheMarksman-ES commented on data/pirate missions smuggling intro.txt:
>log `Accepted a contract from Alexandra Merrin of Liekedeeler Enterprises in exchange for the company to pay off my loan.`

Amazinite commented:
Looks like @Shroomicide is MIA. He hasn't spoken in the Discord server since May of 2019, a month after the most recent message here. As with #4292, if OP returns and is still interested then this PR can be reopened, or someone could open their own PR to maintain this content, but for now I think it's safe to say that this can be closed.

Shroomicide commented:
@Amazinite Sorry for the very late reply. I've been going back and forth on whether I'll have time to continue, but closing is a good call at this point. I've got a significant amount beyond this written, coded, and tested; but it's going to take much more work to bring it to fruition and make it a full arc. I initially opened this pull request as a means to get more feedback and traction, but then I lost my free time, and subsequently, motivation. I worked on it steadily into July and have only picked at it since.

I'd absolutely love to finish this, but I don't want to cause any delays if someone else is interested. I'm going to try to get back into it and dedicate at least a couple hours a week. Hopefully I can get set some reasonable goals for myself and get back into the community as well!

Amazinite commented:
Glad to hear you're at least still interested. Best of luck to you on that. If you're looking for feedback as you develop, then you're probably better off asking around in the Discord server. I know I see people there who will post files with a string of plugin missions or simply post snippets of a mission to get feedback on. I myself will often throw ideas around to see the initial feedback/reactions to them to judge if it's something worth pursuing.
